plugins/tech-insights-backend/src/service/router.ts

```diff
@@ -191,6 +198,18 @@ export async function createRouter<
     );
   });
 
+  router.post('/facts/:namespace/:kind/:name', async (req, res) => {
+    const { namespace, kind, name } = req.params;
+    const { factRetrieverIds }: { factRetrieverIds: string[] } = req.body;
```

Comment:
let's add some validation to verify that `factRetrieverIds` is actually a `string[]`